instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/NettyServerSingletons.java

```diff
@@ -7,21 +7,22 @@
 
 import io.opentelemetry.api.GlobalOpenTelemetry;
 import io.opentelemetry.instrumentation.netty.v4_1.NettyServerTelemetry;
+import io.opentelemetry.instrumentation.netty.v4_1.NettyServerTelemetryBuilder;
+import io.opentelemetry.instrumentation.netty.v4_1.internal.server.NettyServerInstrumenterBuilderUtil;
 import io.opentelemetry.javaagent.bootstrap.internal.AgentCommonConfig;
 
 public final class NettyServerSingletons {
 
   static {
-    SERVER_TELEMETRY =
-        NettyServerTelemetry.builder(GlobalOpenTelemetry.get())
-            .setEmitExperimentalHttpServerEvents(
-                AgentCommonConfig.get().shouldEmitExperimentalHttpServerTelemetry())
-            .setEmitExperimentalHttpServerMetrics(
-                AgentCommonConfig.get().shouldEmitExperimentalHttpServerTelemetry())
-            .setKnownMethods(AgentCommonConfig.get().getKnownHttpRequestMethods())
-            .setCapturedRequestHeaders(AgentCommonConfig.get().getServerRequestHeaders())
-            .setCapturedResponseHeaders(AgentCommonConfig.get().getServerResponseHeaders())
-            .build();
+    NettyServerTelemetryBuilder builder = NettyServerTelemetry.builder(GlobalOpenTelemetry.get());
+    NettyServerInstrumenterBuilderUtil.getBuilderExtractor()
+        .apply(builder)
+        .configure(AgentCommonConfig.get());
+    if (AgentCommonConfig.get().shouldEmitExperimentalHttpServerTelemetry()) {
+      // this logic is only used in agent
+      builder.setEmitExperimentalHttpServerEvents(true);
```

Comment:
is there reason not to expose this to library instrumentation?

Reply:
it is exposed already https://github.com/zeitlinger/opentelemetry-java-instrumentation/blob/6874a63723e7c1b1fc95535811a0a8f4114e4363/instrumentation/netty/netty-4.1/library/src/main/java/io/opentelemetry/instrumentation/netty/v4_1/NettyServerTelemetryBuilder.java#L46

just for javaagent, there is no dedicated property